src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -144,6 +144,20 @@ abstract class RefChecks extends Transform {
           }
         }
       }
+      if (settings.warnDubiousOverload) {
+        val members = clazz.info.members
+        val implicitMembers = members.filter(_.tpe match { case mt: MethodType => mt.isImplicit case _ => false })
+        implicitMembers.foreach { sym =>
```

Comment:
This is just to make me feel smart, but I can trigger two warnings
```scala
scala> class C { def foo(implicit e: String) = 0; def foo(implicit e: Int) = 1; val foo = 0 }
                                                                                    ^
       warning: Usages of value foo will be easy to mistake for calls to def foo(implicit e: Int): Int, which has a single implicit parameter list.
                                                                                    ^
       warning: Usages of value foo will be easy to mistake for calls to def foo(implicit e: String): Int, which has a single implicit parameter list.
```

Reply:
The example of `sortedset.map` makes me think this trivial lint may be worth taking seriously.

Scala 3 even more so with interleaved clauses. The chances increase of "accidentally" adding API that is not an ambiguous overload but "can't be invoked". (There was a long tail about extension methods which "can't" be invoked, so those are real "scare" quotes.)